astropy/units/core.py

```diff
@@ -1638,11 +1638,17 @@ def _recreate_irreducible_unit(cls, names, registered):
     """
     registry = get_current_unit_registry().registry
     if names[0] in registry:
+        # If in local registry return that object.
         return registry[names[0]]
     else:
         unit = cls(names)
         if registered:
+            # If not in local registry but registered in origin registry,
+            # enable unit in local registry.
             get_current_unit_registry().add_enabled_units([unit])
+        else:
+            # If not registered at all, just return the unit.
+            return unit
```

Comment:
Very strange that this was missing! But I think it should also return the unit if `registered` is true (ie., you can just remove the `else` and deindent this).